src/views/TreasuryDashboard/TreasuryDashboard.tsx

```diff
@@ -100,11 +101,40 @@ const MetricsDashboard = () => (
   </>
 );
 
+enum DashboardTab {
+  OP = "/dashboard/olympuspro",
+  Proteus = "/dashboard/proteus",
+  Revenue = "/dashboard/revenue",
+  Treasury = "/dashboard/treasury",
+}
+
 const TreasuryDashboard = memo(() => {
+  const location = useLocation();
   const [view, setView] = useState(0);
   const isSmallScreen = useMediaQuery("(max-width: 650px)");
   const isVerySmallScreen = useMediaQuery("(max-width: 379px)");
 
+  useEffect(() => {
+    switch (location.pathname) {
+      case DashboardTab.Treasury:
+        setView(1);
+        break;
+      case DashboardTab.Revenue:
+        setView(2);
+        break;
+      case DashboardTab.OP:
+        setView(3);
+        break;
+      case DashboardTab.Proteus:
+        setView(4);
+        break;
+      default:
+        setView(0);
+        break;
+    }
+  }, [location]);
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
instead of using useEffect, we can pass a prop to TreasuryDashboard to determine which view should be set. 
See examples of how this was done with Wallet.

Reply:
Yeah this is a fair point and to be honest I was going to go down the route of adding specific routes for each tab but opted out of it in the end.

Will rework this today and update the PR, thank you for the feedback, much appreciated.